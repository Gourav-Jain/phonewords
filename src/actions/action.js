/**
 * Action Types
 */
export const PHONE_WORD_REQS = "PHONE_WORD_REQS";
export const PHONE_WORD_RESP = "PHONE_WORD_RESP";
export const PHONE_WORD_FAIL = "PHONE_WORD_FAIL";

/**
 * Action Creator
 */
export const setPhonewordRequest = (payload) => ({
  type: PHONE_WORD_REQS,
  payload,
});
export const setPhonewordResponse = (payload) => ({
  type: PHONE_WORD_RESP,
  payload,
});
export const setPhonewordFailer = (payload) => ({
  type: PHONE_WORD_FAIL,
  payload,
});
