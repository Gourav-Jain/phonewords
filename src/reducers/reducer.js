import {
  PHONE_WORD_REQS,
  PHONE_WORD_RESP,
  PHONE_WORD_FAIL,
} from "../actions/action";
import initialState from "../store/initialState";

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case PHONE_WORD_REQS:
      return {
        ...state,
        ...action.payload,
      };
    case PHONE_WORD_RESP:
      return { ...state, ...action.payload };
    case PHONE_WORD_FAIL:
      return { ...state, ...action.payload };
    default:
      return state;
  }
};
