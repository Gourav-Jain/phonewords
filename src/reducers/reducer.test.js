import { reducer as phonewordReducer } from "./reducer";
import * as actions from "../actions/action";

describe("Phoneword reducer", () => {
  test.each`
    name                                 | reducerFn                 | state   | mockData
    ${"sucessfull setPhonewordRequest"}  | ${"setPhonewordRequest"}  | ${null} | ${""}
    ${"sucessfull setPhonewordResponse"} | ${"setPhonewordResponse"} | ${null} | ${""}
    ${"sucessfull setPhonewordFailer"}   | ${"setPhonewordFailer"}   | ${null} | ${""}
  `("should return $name", ({ reducerFn, state, mockData }) => {
    const action = reducerFn ? actions[reducerFn](mockData) : {};
    expect(phonewordReducer(state, action)).toMatchSnapshot();
  });
});
