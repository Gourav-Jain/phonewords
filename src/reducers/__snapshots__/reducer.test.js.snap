// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phoneword reducer should return sucessfull setPhonewordFailer 1`] = `Object {}`;

exports[`Phoneword reducer should return sucessfull setPhonewordRequest 1`] = `Object {}`;

exports[`Phoneword reducer should return sucessfull setPhonewordResponse 1`] = `Object {}`;
